### ISSUE TITLE
Is it possible to sort only once?

### ISSUE THREAD

**Issue by 1341290135** [objectbox/objectbox-java#775]
Query<User> query = userBox.query().order(User_.firstName).contain(User_.firstName, "").parameterAlias("name").build;

step1:
List<User> joes = query.setParameter("name", "J").find(0,19);

step2:
List<User> joes = query.setParameter("name", "J").find(20,39);

I found that the step2 was still reordering again, so that means sort every time, and the speed did not increase.

Is there a way to sort once, and later queries can use the previous sort, without reordering, to speed up the process?

**Comment by greenrobot-team** [comment]
There might be room for internal optimizations. But think about this case: what happens if a new `User` is put right between step1 and step2?

**Comment by 1341290135** [comment]
I think it might be like this: if new user data modified, order will be recalculated, if not, then there is no need to execute order every time.

The problem now is that order is a very time-consuming operation (about 160K in database, about 1.5s need), and query are mostly paginated, requiring only part of the data at a time. If order can be recalculated without reading one page (such as 20 data) at a time, the speed will increase greatly.



Or is there any better way to deal with this application? Please give me some advice. Thank you very much.

**Comment by greenrobot-team** [comment]
Sorry, I should have been more clear: there is nothing you can do. This is up to future internal optimizations by us.

**Comment by greenrobot-team** [comment]
As this is about internal optimization, which is an ongoing process, there is nothing to track here. Closing.

Feel free to create a new issue regarding reproducible performance problems with the latest release.
